test/unit/theory/theory_sets_type_rules_white.cpp

```diff
@@ -0,0 +1,87 @@
+/*********************                                                        */
+/*! \file theory_sets_type_rules_white.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Aina Niemetz, Mudathir Mohamed
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief White box testing of sets typing rules
+ **/
+
+#include "expr/dtype.h"
+#include "test_api.h"
+#include "test_node.h"
+
+namespace CVC4 {
+
+using namespace CVC4::api;
+
+namespace test {
+
+class TestTheoryWhiteSetsTypeRuleApi : public TestApi
+{
+};
+class TestTheoryWhiteSetsTypeRuleInternal : public TestNode
+{
+};
+
+TEST_F(TestTheoryWhiteSetsTypeRuleInternal, is_comparable_to)
+{
+  TypeNode setRealType = d_nodeManager->mkSetType(d_nodeManager->realType());
+  TypeNode setIntType = d_nodeManager->mkSetType(d_nodeManager->integerType());
+  ASSERT_FALSE(setIntType.isComparableTo(setRealType));
+  ASSERT_FALSE(setIntType.isSubtypeOf(setRealType));
+  ASSERT_FALSE(setRealType.isComparableTo(setIntType));
+  ASSERT_FALSE(setRealType.isComparableTo(setIntType));
+}
+
+TEST_F(TestTheoryWhiteSetsTypeRuleApi, singleton_term)
+{
+  Sort realSort = d_solver.getRealSort();
+  Sort intSort = d_solver.getIntegerSort();
+  Term emptyReal = d_solver.mkEmptySet(d_solver.mkSetSort(realSort));
+  Term integerOne = d_solver.mkInteger(1);
+  Term realOne = d_solver.mkReal(1);
+  Term singletonInt = d_solver.mkTerm(api::SINGLETON, integerOne);
+  Term singletonReal = d_solver.mkTerm(api::SINGLETON, realOne);
+  // (union
+  //    (singleton (singleton_op Int) 1)
+  //    (as emptyset (Set Real)))
+  ASSERT_THROW(d_solver.mkTerm(UNION, singletonInt, emptyReal),
+               CVC4ApiException);
+  // (union
+  //    (singleton (singleton_op Real) 1)
+  //    (as emptyset (Set Real)))
+  ASSERT_NO_THROW(d_solver.mkTerm(UNION, singletonReal, emptyReal));
+}
+
+TEST_F(TestTheoryWhiteSetsTypeRuleInternal, singleton_node)
+{
+  Node singletonInt =
+      d_nodeManager->mkConst(SingletonOp(d_nodeManager->integerType()));
+  Node singletonReal =
+      d_nodeManager->mkConst(SingletonOp(d_nodeManager->realType()));
+  Node intConstant = d_nodeManager->mkConst(Rational(1));
+  Node realConstant = d_nodeManager->mkConst(Rational(1, 5));
+  // (singleton (singleton_op Real) 1)
+  ASSERT_NO_THROW(
+      d_nodeManager->mkSingleton(d_nodeManager->realType(), intConstant));
+  // (singleton (singleton_op Int) (/ 1 5))
+  // This fails now with the assertions.
```

Comment:
with `ASSERT_DEATH` in google test, this comment is no longer needed, and can be removed